xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1066,6 +1066,12 @@ int CGUIDialogVideoInfo::ManageVideoItem(const CFileItemPtr &item)
         result = RemoveItemsFromTag(item);
         break;
 
+      case CONTEXT_BUTTON_MARK_WATCHED:
+      case CONTEXT_BUTTON_MARK_UNWATCHED:
+        database.MarkAsWatched(*item, (button == (CONTEXT_BUTTON)CONTEXT_BUTTON_MARK_WATCHED));
```

Comment:
I made it not go through the `CVideoLibraryQueue` as we're only handling one item here.